src/main/java/com/amazon/dlic/auth/http/saml/HTTPMetadataResolver.java

```diff
@@ -0,0 +1,157 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package com.amazon.dlic.auth.http.saml;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.net.URI;
+import java.net.URISyntaxException;
+import java.util.Timer;
+
+import org.apache.hc.client5.http.classic.HttpClient;
+import org.apache.hc.client5.http.classic.methods.HttpGet;
+import org.apache.hc.client5.http.protocol.HttpClientContext;
+import org.apache.hc.core5.http.ClassicHttpResponse;
+import org.apache.hc.core5.http.Header;
+import org.apache.hc.core5.http.HttpException;
+import org.apache.hc.core5.http.HttpHeaders;
+import org.apache.hc.core5.http.io.entity.EntityUtils;
+import org.apache.hc.core5.http.HttpStatus;
+
+import net.shibboleth.utilities.java.support.resolver.ResolverException;
+import org.opensaml.saml.metadata.resolver.impl.AbstractReloadingMetadataResolver;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class HTTPMetadataResolver extends AbstractReloadingMetadataResolver {
```

Comment:
I see code patterns around the logging that are unique and distinct from the security codebase, was this class's contents copied from another implementation?

Reply:
You're right, I followed suggestion no. 1 from https://github.com/opensearch-project/security/issues/2932#issue-1788296478
Switched to LogManager.getLogger() for uniformity although i noticed package org.slf4j.LoggerFactory is used in one more place: CleanerFactory.class